audio/openal-soft-devel/Portfile

```diff
@@ -0,0 +1,110 @@
+
+name                    openal-soft-devel
+conflicts               openal-soft
+github.setup            kcat openal-soft 6790234a907e23f91f5d1eb870fe23c3b42d3496
+version                 2024.12.11
+revision                0
+checksums               rmd160  fda1dc0064641c54d77fb032cbd8d714e4c08b06 \
+                        sha256  c5edbda4cf6d7eeb1d45af0755f416f183d8032d63717a37ddf476aa4064d6f1 \
+                        size    1207281
+github.tarball_from     archive
+
+categories              audio devel
+maintainers             nomaintainer
+license                 LGPL-2
+
+description             Cross-platform software implementation of the OpenAL \
+                        3D audio API
+long_description        OpenAL Soft is a ${description}. It's forked from \
+                        the open-sourced Windows version available originally \
+                        from the SVN repository at openal.org. OpenAL \
+                        provides capabilities for playing audio in a virtual \
+                        3D environment. Distance attenuation, doppler shift, \
+                        and directional sound emitters are among the features \
+                        handled by the API. More advanced effects, including \
+                        air absorption, occlusion, and environmental reverb, \
+                        are available through the EFX extension. It also \
+                        facilitates streaming audio, multi-channel buffers, \
+                        and audio capture.
+
+homepage                https://openal-soft.org
+
+patchfiles-append       0001-coreaudio.cpp-restore-fallbacks-for-10.5.patch
+
+depends_build-append    path:bin/pkg-config:pkgconfig
+
+depends_lib-append      port:libmysofa \
+                        port:libsndfile \
+                        port:zlib
+
+compiler.cxx_standard           2017
+
+compiler.thread_local_storage   yes
+
+# error: inline declaration of 'configValue<...>' follows non-inline definition
+# https://github.com/kcat/openal-soft/issues/703
+compiler.blacklist-append {clang < 900}
+
+configure.args-append   -DALSOFT_EXAMPLES=OFF \
+                        -DALSOFT_UTILS=ON \
+                        -DALSOFT_NO_CONFIG_UTIL=ON \
+                        -DALSOFT_REQUIRE_JACK=OFF \
+                        -DALSOFT_BACKEND_JACK=OFF \
+                        -DALSOFT_REQUIRE_PORTAUDIO=OFF \
+                        -DALSOFT_BACKEND_PORTAUDIO=OFF \
+                        -DALSOFT_REQUIRE_PULSEAUDIO=OFF \
+                        -DALSOFT_BACKEND_PULSEAUDIO=OFF \
+                        -DALSOFT_REQUIRE_SDL2=OFF \
+                        -DALSOFT_BACKEND_SDL2=OFF
+
+platform darwin {
+    # https://github.com/kcat/openal-soft/issues/1066
+    if {${os.major} <= 16} {
+        if {[string match *clang* ${configure.compiler}]} {
+            configure.cxxflags-append \
+                        -fno-aligned-allocation
+        }
+    }
```

Comment:
It is absolutely not safe to pass `-fno-aligned-allocation` just to silence an error, because it will result in incorrect behaviour within openal-soft.

What's needed here is to provide a set of working aligned allocator functions (based on `posix_memalign` or something, like openal-soft used to use) with the right names, and pass `-faligned-allocation` to silence the error.

Reply:
I can drop this chunk altogether, the port will just not build on affected systems, but for a -devel subport this is not a big concern. My idea was to have it exactly to be able to test stuff without affecting the main `openal-soft` port.

How does it work with gcc now, do you know? If it was `aligned_alloc`, gcc build would fail without a fallback, but it does not fail (no flags or hacks needed).

Reply:
I assume it works with gcc 14 because libgcc provides an implementation of the aligned allocators, but libc++ does not on macOS <10.13.

Just a note for anyone that wants to look at this further: `posix_memalign` is only supported on >=10.6

Reply:
> Just a note for anyone that wants to look at this further: posix_memalign is only supported on >=10.6

Yes, but `legacysupport` supplies it for < 10.6, AFAIK.